File: app/src/main/java/co/tpcreative/supersafe/ui/lockscreen/LockScreenPresenter.java
package co.tpcreative.supersafe.ui.lockscreen;
import co.tpcreative.supersafe.common.presenter.BaseView;
import co.tpcreative.supersafe.common.presenter.Presenter;
import co.tpcreative.supersafe.model.EnumPinAction;
import co.tpcreative.supersafe.model.EnumStatus;

public class LockScreenPresenter extends Presenter<BaseView<EnumPinAction>>{
    public LockScreenPresenter(){
    }

    public void onChangeStatus(EnumStatus status,EnumPinAction action){
        BaseView view = view();
        view.onSuccessful("Successful",status,action);
    }
}

File: app/src/main/java/co/tpcreative/supersafe/ui/lockscreen/LockScreenPresenter.kt
package co.tpcreative.supersafe.ui.lockscreen
import co.tpcreative.supersafe.common.presenter.BaseView
import co.tpcreative.supersafe.common.presenter.Presenter
import co.tpcreative.supersafe.model.EnumPinAction
import co.tpcreative.supersafe.model.EnumStatus

class LockScreenPresenter : Presenter<BaseView<EnumPinAction>>() {
    fun onChangeStatus(status: EnumStatus?, action: EnumPinAction?) {
        val view: BaseView<EnumPinAction>? = view()
        view?.onSuccessful("Successful", status, action)
    }
}